src/main/java/com/blamejared/crafttweaker/impl/recipes/Replacer.java

```diff
@@ -0,0 +1,113 @@
+import com.blamejared.crafttweaker.api.item.IIngredient;
+import com.blamejared.crafttweaker.api.managers.IRecipeManager;
+import com.blamejared.crafttweaker.api.recipes.IRecipeHandler;
+import com.blamejared.crafttweaker.api.recipes.ReplacementRule;
+import com.blamejared.crafttweaker.impl.actions.recipes.ActionReplaceRecipe;
+import com.blamejared.crafttweaker.impl.brackets.RecipeTypeBracketHandler;
+import com.blamejared.crafttweaker.impl.recipes.wrappers.WrapperRecipe;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import com.mojang.datafixers.util.Pair;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.util.registry.Registry;
+import org.openzen.zencode.java.ZenCodeType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@ZenRegister
+@ZenCodeType.Name("crafttweaker.api.recipes.Replacer")
+@Document("vanilla/api/recipes/Replacer")
+public final class Replacer {
+   
+    private final Collection<IRecipeManager> targetedManagers;
+    private final Collection<? extends IRecipe<?>> targetedRecipes;
+    private final List<ReplacementRule> rules;
+    
+    private Replacer(final Collection<? extends IRecipe<?>> recipes, final Collection<IRecipeManager> managers) {
+        this.targetedManagers = Collections.unmodifiableCollection(managers);
+        this.targetedRecipes = Collections.unmodifiableCollection(recipes);
+        this.rules = new ArrayList<>();
+    }
+    
+    @ZenCodeType.Method
+    public static Replacer forRecipes(final WrapperRecipe... recipes) {
+        if (recipes.length <= 0) {
+            throw new IllegalArgumentException("Unable to create a replacer without any targeted recipes");
+        }
+        return new Replacer(Arrays.stream(recipes).map(WrapperRecipe::getRecipe).collect(Collectors.toList()), Collections.emptyList());
+    }
+    
+    @ZenCodeType.Method
+    public static Replacer forTypes(final IRecipeManager... managers) {
+        if (managers.length <= 0) {
+            throw new IllegalArgumentException("Unable to create a replacer without any targeted recipe types");
+        }
+        return new Replacer(Collections.emptyList(), Arrays.asList(managers));
+    }
+    
+    @ZenCodeType.Method
+    public static Replacer forAllTypes() {
+        return new Replacer(Collections.emptyList(), RecipeTypeBracketHandler.getManagerInstances());
+    }
+    
+    @ZenCodeType.Method
+    public Replacer replace(final IIngredient from, final IIngredient to) {
+        // TODO("Avoid trivial replacement rules -- check if from == to")
+        if (from.contains(to) && to.contains(from)) return this;
+        this.rules.add(new ReplacementRule(from, to));
+        return this;
+    }
+    
```

Comment:
Maybe add a method 
```java
public Replacer replace(ReplacementRule rule) {
    this.rules.add(rule);
    return this;
}
```
Just so that other mods could ZenExpand the replacer class and add custom Replacer rules.
Like, for example, Mekanism could do something like
```less
Replacer.forTypes(craftingTable).replaceGas(<gas:oxygen> * 200, <gas:nitrogen> * 100).execute();
```
Or something (I have actually no idea if there are mek recipes that need a container with a gas present lol)
Or if you exposed the Rules to ZC, then even players could get creative lol.